File: src/RoutinePlanner.Api/Models/Goal.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace RoutinePlanner.Api.Models;

public class Goal
{
    public int Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int Year { get; set; }
    public Season Season { get; set; }
    public SubPeriod SubPeriod { get; set; }
    // Если задано вручную — используется это значение вместо суммы по GoalItems
    public decimal? ManualCost { get; set; }
    public bool IsCompleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public List<GoalItem> Items { get; set; } = new();

    // Не хранится в базе — вычисляется на лету
    public decimal TotalCost => ManualCost ?? Items.Sum(i => i.Cost);
}

File: routine-planner-web/src/types/goal.ts
export enum Season {
    Winter = 0,
    Spring = 1,
    Summer = 2,
    Autumn = 3
}

export enum SubPeriod {
    Early = 0,
    Mid = 1,
    Late = 2
}

export interface GoalItemDto {
    id: number;
    title: string;
    cost: number;
    isCompleted: boolean;
    createdAt: string;
}

export interface GoalDto {
    id: number;
    title: string;
    description?: string;
    year: number;
    season: Season;
    subPeriod: SubPeriod;
    totalCost: number;
    isCompleted: boolean;
    createdAt: string;
    items: GoalItemDto[];
}